data_processing/wdl/workflows/hifi_qc_workflow.wdl

```diff
@@ -2,50 +2,137 @@ version 1.0
 
 import "../tasks/read_stats.wdl" as read_stats_wf
 import "../tasks/ntsm.wdl" as ntsm_check
+import "../tasks/methylationCheck.wdl" as meth_check  
 
 workflow hifi_qc_wf {
     input {
-        Array[File] hifi_reads
+        File hifi_reads
         Array[File] other_reads
         String sample_id
+        Boolean perform_methylation_check = false  # Optional input to perform methylation check
     }
 
     parameter_meta {
         hifi_reads: "Reads to QC. Can be fastq, fastq.gz, or bam."
         other_reads: "Trusted reads to compare against to check for sample swaps. Can be fastq, fastq.gz, or bam."
         sample_id: "Id to use for output naming."
+        perform_methylation_check: "Set to true to perform methylation check on BAM files."
     }
 
     meta {
-        author: "Julian Lucas"
-        email: "juklucas@ucsc.edu"
+        author: "Julian Lucas, Andrew Blair"
+        email: "juklucas@ucsc.edu, apblair@ucsc.edu"
     }
     
     call read_stats_wf.runReadStats as read_stats {
         input: 
-            reads          = hifi_reads,
-            identifier     = sample_id
+            reads = [hifi_reads],
+            identifier = sample_id
     }
 
     call ntsm_check.ntsm_workflow as ntsm_wf {
         input:
-            input_reads_1  = hifi_reads,
-            input_reads_2  = other_reads,
-            sample_id      = sample_id,
-            read_1_type    = "hifi",
-            read_2_type    = "other"
+            input_reads_1 = [hifi_reads],
+            input_reads_2 = other_reads,
+            sample_id = sample_id,
+            read_1_type = "hifi",
+            read_2_type = "other"
     }
 
+    # Optional methylation check
+    if (perform_methylation_check) {
+        call meth_check.methylationWorkflow as methylation {
+            input:
+                bam_files = [hifi_reads],
+                sample_id = sample_id
+        }
+    }
+    
+
+    call summarize_hifi_qc {
+        input:
+            ntsm_output = ntsm_wf.ntsm_eval_out,
+            readstat_report = read_stats.ReadStatsReport,
+            file_name = sample_id,
+            methylation_report = select_first([if (perform_methylation_check) then methylation.methylation_outputs else []])
+    }
 
     output {
         ## readstats output
-        File hifi_readstat_tar    = read_stats.ReadStatsTarball 
+        File hifi_readstat_tar = read_stats.ReadStatsTarball 
         File hifi_readstat_report = read_stats.ReadStatsReport
-
+        
         ## ntsm output
-        File hifi_ntsm_counts     = ntsm_wf.ntsm_count_1
-        File ext_ntsm_counts      = ntsm_wf.ntsm_count_2
-        File ntsm_eval            = ntsm_wf.ntsm_eval_out
+        File hifi_ntsm_counts = ntsm_wf.ntsm_count_1
+        File ext_ntsm_counts = ntsm_wf.ntsm_count_2
+        File ntsm_eval = ntsm_wf.ntsm_eval_out
+        File hifi_qc_summary = summarize_hifi_qc.summary_file
```

Comment:
I'd reorg like so:
```
 File hifi_qc_summary = summarize_hifi_qc.summary_file

## methylation check
Array[File]? methylation_report = methylation.methylation_outputs
```
I don't *think* this needs the if/else logic since you have the type set to optional

Reply:
I'd put the qc_summary file at the list of the list, too.